src/measurement/idblock.rs

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Functions to use to calculate the ID-BLOCK and the AUTH-BLOCK.
+
+use bincode;
+use openssl::{ec::EcKey, pkey::Private, sha::sha384};
+use std::{
+    convert::{TryFrom, TryInto},
+    fs::File,
+    io::Read,
+    path::PathBuf,
+};
+
+use crate::{
+    error::IdBlockError,
+    measurement::idblock_types::{
+        FamilyId, IdAuth, IdBlock, IdBlockLaunchDigest, IdMeasurements, ImageId, SevEcdsaPubKey,
+        SevEcdsaSig, CURVE_P384_NID,
+    },
+};
+
+/// Generate an AUTH-BLOCK using 2 EC P-384 keys and an already calculated ID-BlOCK
+pub fn gen_id_auth_block(
+    id_block: &IdBlock,
+    id_key_file: PathBuf,
+    author_key_file: PathBuf,
+) -> Result<IdAuth, IdBlockError> {
+    let id_ec_priv_key = load_priv_key_from_pem(id_key_file)?;
+    let id_ec_pub_key = SevEcdsaPubKey::try_from(&id_ec_priv_key)?;
+    let id_sig = SevEcdsaSig::try_from((
+        id_ec_priv_key,
+        bincode::serialize(id_block)
+            .map_err(|e| IdBlockError::BincodeError(*e))?
+            .as_slice(),
+    ))?;
+
+    let author_ec_priv_key = load_priv_key_from_pem(author_key_file)?;
+    let author_pub_key = SevEcdsaPubKey::try_from(&author_ec_priv_key)?;
+    let author_sig = SevEcdsaSig::try_from((
+        author_ec_priv_key,
+        bincode::serialize(&id_ec_pub_key)
+            .map_err(|e| IdBlockError::BincodeError(*e))?
+            .as_slice(),
+    ))?;
+
+    Ok(IdAuth::new(
+        None,
+        None,
+        id_sig,
+        id_ec_pub_key,
+        author_sig,
+        author_pub_key,
+    ))
+}
+
+///Read a pem key file and return a private EcKey.
+/// Key has to be an EC P-384 key.
+pub fn load_priv_key_from_pem(path: PathBuf) -> Result<EcKey<Private>, IdBlockError> {
```

Comment:
Do we foresee any requests to want DER support?

Reply:
yes :(

Begin ec private key

Reply:
----- BEGIN PUBLIC KEY -----

You mean this? ^^